src/katgpucbf/fgpu/main.py

```diff
@@ -266,6 +330,15 @@ def parse_args(arglist: Sequence[str] | None = None) -> argparse.Namespace:
             parser.error("Live source requires --src-interface")
     if len(args.dst) % len(args.dst_interface) != 0:
         parser.error("number of destinations must be divisible by number of destination interfaces")
+
+    # Convert narrowband from NarrowbandOutputDict to NarrowbandOutput
+    for output in args.narrowband:
+        if "taps" not in output:
+            output["taps"] = args.taps
+        if "w_cutoff" not in output:
+            output["taps"] = args.w_cutoff
+    args.narrowband = [NarrowbandOutput(*output) for output in args.narrowband]
```

Comment:
After originally typing this comment up, perhaps it's best if we just chat for a few minutes for you to explain the logic to me (please).

It's likely
* my ipython hackery to understand this, 
* coupled with my ~incomplete understanding of what's meant to happen,
* and/or what it means to iterate over a `TypedDict`,

are all contributing to my general confusion about this line.

In a nutshell, 
* I'm not sure if this is meant to yield a single `NarrowbandOutput` instance, and
* I'm thrown by it seeming like it creates a separate instance for each `output` it expands/flattens.